src/logging/CGlogLoggerAdaptor.cpp

```diff
@@ -47,9 +48,7 @@ void CGlogLoggerAdaptor::Write(bool force_flush,
 #endif
     
     if (_mylogger && message_len > 0) {
-        if (_mutex) {
-            _mutex->lock();
-        }
+        _mutex.lock();
```

Comment:
You should really use a `std::lock_guard` here instead of locking/unlocking manually.